numba/unicode.py

```diff
@@ -1361,6 +1361,26 @@ def impl(a):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L11896-L11925    # noqa: E501
+@overload_method(types.UnicodeType, 'isspace')
+def unicode_isspace(data):
```

Comment:
I think implementing this permits e.g. https://github.com/numba/numba/pull/4835/files#diff-a21787256ad779b4c4c5c99915c71025R737 to be refactored and removal of https://github.com/numba/numba/pull/4835/files#diff-a21787256ad779b4c4c5c99915c71025R385-R417 ? If you agree this change should go into this PR to prevent duplication of functionality.

Reply:
I think we need to replace `is_whitespace = _is_whitespace(code_point)` with `is_whitespace = _PyUnicode_IsSpace(code_point)` in `str.split` implementation and then remove function `_is_whitespace()`. Let me make the change in this PR.

Reply:
Done!